prov/shm/src/smr_av.c

```diff
@@ -220,7 +220,7 @@ int smr_av_open(struct fid_domain *domain, struct fi_av_attr *attr,
 	if (!smr_av)
 		return -FI_ENOMEM;
 
-	util_attr.addrlen = NAME_MAX + 1;
+	util_attr.addrlen = NAME_MAX;
```

Comment:
@aingerson - can you confirm the correct addrlen for shm.  NAME_MAX excludes a null terminating character.  Is the correct addrlen NAME_MAX + 1, or do you exclude the null character from the address?  The other pieces of the shm code don't make this clear.

Reply:
This fix is correct. It looks like this line was incorrectly changed to add +1 recently in commit 51c2f96c634bca44c994a09cc406678a1c03cdc5 by mistake

Reply:
I understand that this is reverting the change in the prior commit.  My question is whether NAME_MAX alone should be the right value for the shm addrlen, since NAME_MAX is defined to exclude the null terminating character of a string.

Reply:
Ahh yes that is confusing then. I was using it as the size of the addr which does include null character. That is how it is used the rest of the provider but is NAME_MAX is defined to exclude it, then we should probably change the rest of the implementation to match that or else redefine the shm addrlen to not use NAME_MAX and avoid confusion since the shm addrlens are all data lengths, not string lengths. I'm more tempted to the second option since it's a different usage of the term entirely and aligns the data better.

Reply:
I agree using a different value for shm makes more sense.